File: leetcode/0013/romanToInt.swift
func romanToInt(_ s: String) -> Int {
    let hashTable:[Character: Int] = ["I":1, "V":5, "X":10, "L":50, "C":100, "D":500, "M":1000]
    var sum = 0
    var last = 0
    for char in s {
        if let cur = hashTable[char] {
            if (cur > last) {
                sum = sum - 2*last + cur
            } else {
                sum = sum + cur
            }
            last = cur
        }
        
    }
    return sum
}

print(romanToInt("XC"))

File: leetcode/0013/romanToInt.py
def romanToInt(s: str) -> int:
    d = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000}
    sum = 0
    num = len(s)
    for i in range(num):
        if i < num - 1 and d[s[i]] < d[s[i + 1]]:
            sum -= d[s[i]]
        else:
            sum += d[s[i]]
    return sum

print(romanToInt('XCIII'))